File: helfomat-core/src/main/java/de/helfenkannjeder/helfomat/core/organization/event/OrganizationEditNameEvent.java
package de.helfenkannjeder.helfomat.core.organization.event;

import de.helfenkannjeder.helfomat.core.organization.Organization;
import de.helfenkannjeder.helfomat.core.organization.OrganizationEventVisitor;
import de.helfenkannjeder.helfomat.core.organization.OrganizationId;

/**
 * @author Valentin Zickner
 */
@SuppressWarnings({"WeakerAccess", "CanBeFinal"})
public class OrganizationEditNameEvent extends OrganizationEditEvent {
    private String name;

    protected OrganizationEditNameEvent() {
    }

    public OrganizationEditNameEvent(OrganizationId organizationId, String name) {
        super(organizationId);
        this.name = name;
    }

    public String getName() {
        return name;
    }

    @Override
    public Organization.Builder applyOnOrganizationBuilder(Organization.Builder organization) {
        return organization.setName(name);
    }

    @Override
    public <T> T visit(OrganizationEventVisitor<T> visitor) {
        return visitor.visit(this);
    }
}

File: helfomat-core/src/main/kotlin/de/helfenkannjeder/helfomat/core/organization/event/OrganizationEditNameEvent.kt
package de.helfenkannjeder.helfomat.core.organization.event

import de.helfenkannjeder.helfomat.core.organization.Organization
import de.helfenkannjeder.helfomat.core.organization.OrganizationEventVisitor
import de.helfenkannjeder.helfomat.core.organization.OrganizationId

/**
 * @author Valentin Zickner
 */
data class OrganizationEditNameEvent(
    override val organizationId: OrganizationId,
    val name: String
) : OrganizationEditEvent(organizationId) {

    override fun applyOnOrganizationBuilder(organizationBuilder: Organization.Builder?, strictMode: Boolean): Organization.Builder? {
        return organizationBuilder?.setName(name)
    }

    override fun <T> visit(visitor: OrganizationEventVisitor<T>): T {
        return visitor.visit(this)
    }

}